lib/utils/render-status.html

```diff
@@ -32,7 +32,10 @@
   }
 
   function flushQueue(queue) {
-    while (queue.length) {
+    const max = queue.length;
+    let i=0;
+    while (queue.length && i < max) {
+      i++;
```

Comment:
I assume this change is from a realization that calling `RenderStatus.afterNextRender` from within an `afterNextRender` callback should _not_ go in the current turn, right?  However, seems like `beforeNextRender` called within a `beforeNextRender` _should_ go in the current turn, and this breaks it since `flushQueue` is used for both?

Also, keeping the while construction makes it hard to grok; this would read cleaner: `for (let i=0, max=queue.length; i<max; i++)`.